### DISCUSSION TITLE
图片上方与正文间距问题

### DISCUSSION THREAD

**Discussion by Jiuyoung** [note286/xduts#57] [Q&A] [answered]
根据word模板的要求，图片上方与正文应空一行，目前的实现好像不够一行的高度，不知道是不是我配置的问题

![image](https://user-images.githubusercontent.com/25689472/222881504-56a6ce1f-6c78-449c-90b2-2bc9747876f6.png)

**Comment by note286** [comment]
尽量别用`[h]`，不是什么好习惯。

**Reply by Jiuyoung** [reply]
一般情况都是默认不加的，但是有部分图片需要这样，看起来有点挤

**Reply by note286** [reply]
我设一下吧，一会提交。

**Reply by sikouhjw** [reply]
> 一般情况都是默认不加的，但是有部分图片需要这样，看起来有点挤

事实上，部分习惯阅读英文文献的老师会认为，图片得放 top『才对』

**Reply by note286** [reply]
> 事实上，部分习惯阅读英文文献的老师会认为，图片得放 top『才对』

我脑海中，只有试卷是直接放在旁边，不需要caption。都有序号和caption了，还放旁边的意义是什么。

**Reply by Jiuyoung** [reply]
> 是的是的  有时候一页都是图片，目前比较小白，不太会排版，就强制设置到当前位置😂

一整页都是图表是允许的。

**Comment by sikouhjw** [comment]
浮动体的间距设置十分复杂，可以参考
https://github.com/sikouhjw/gdutthesis/blob/19b068e9ef1685443204040206dace634b927228/gdutthesis.cls#L1127-L1153
设置

**Comment by note286** [comment, marked as answer]
你手动在`[h]`的浮动体后加`\nointerlineskip`吧，例如：

```latex
\begin{figure}[h]
xxx
\end{figure}
\nointerlineskip
```

至于间距调大点，你可以：

```latex
\setlength{\intextsep}{20pt}
\begin{document}
```

**Reply by Jiuyoung** [reply]
好的 感谢👍

**Comment by eutopia-61** [comment]
我按照这种方法设置，但是如果碰到图名和标题连着的时候，间距好像又大了点 :see_no_evil:
![image](https://user-images.githubusercontent.com/67130065/224473597-56af71c8-6486-408f-aef4-8a04cf153dec.png)

**Reply by note286** [reply]
目前不需要额外设置，请使用最新版xduts。

**Reply by eutopia-61** [reply]
好的，感谢

**Reply by note286** [reply]
好像确实有点问题，我去修修。

**Reply by eutopia-61** [reply]
对，好像上方不够一行，:see_no_evil:
![image](https://user-images.githubusercontent.com/67130065/224474364-bbce1f55-294c-44d0-871f-3a55ee5010d5.png)

**Reply by note286** [reply]
不要用`[h]`。

**Reply by eutopia-61** [reply]
有些图感觉需要用上h，不然跑在小节后面去了

**Reply by note286** [reply]
你要是非要用，自己看`caption`文档调节吧。

**Reply by sikouhjw** [reply]
> 有些图感觉需要用上h，不然跑在小节后面去了

这是你的认知错误，浮动体在哪里『都可以』。多看英文文献，图片在参考文献后都没有问题

**Reply by note286** [reply]
`{\ctexset{section/beforeskip={0pt}}\section{title}}`，先将就用一下，最近没啥时间。

**Reply by eutopia-61** [reply]
好的，感谢
